vortex-sampling-compressor/src/sampling_compressor.rs

```diff
@@ -159,10 +159,23 @@ impl<'a> SamplingCompressor<'a> {
         Ok(compressed)
     }
 
-    pub fn compress_validity(&self, validity: Validity) -> VortexResult<Validity> {
+    pub fn compress_validity(
+        &self,
+        validity: Validity,
+        like: Option<&CompressionTree<'a>>,
+    ) -> VortexResult<(Validity, Option<CompressionTree<'a>>)> {
         match validity {
-            Validity::Array(a) => Ok(Validity::Array(self.compress(&a, None)?.into_array())),
-            a => Ok(a),
+            Validity::Array(a) => {
+                let comp = self.named("validity").compress(&a, like)?;
+                let arr = Validity::Array(comp.clone().into_array());
+                let val = arr.to_logical(comp.len()).into_validity();
+                if val.as_array().is_some() {
+                    Ok((val, comp.into_path()))
```

Comment:
If the validity array is a constant-like then it will not be included in the encoding, but will have a compressed path. Check for this here.